### ISSUE TITLE
Mongo not connecting with username and password for authentication on Linux

### ISSUE THREAD

**Issue by sujaykakkad** [OpenKitten/MongoKitten#124]
```sh
Authenticating the connection to MongoKitten.Database<mongodb://localhost:27017/samespace_alpha>    [Main:Convenience.swift:info] org.openkitten.mongokitten[0m
[0;31m13:50:44 [Error] Authentication for MongoDB user root with SASL failed against Optional("admin") because of the following error    [Main:Convenience.swift:error] org.openkitten.mongokitten[0m
[0;31m13:50:44 [Error] {"ok":0.0,"errmsg":"Authentication failed.","code":18,"codeName":"AuthenticationFailed"}    [Main:Database+Authentication.swift:challenge(with:using:usingConnection:)] 
```

**Comment by sujaykakkad** [comment]
```swift
let server = try Server(hostname: "localhost", port: 27017, authenticatedAs: MongoCredentials(username: "username", password: "password", database: "admin"))
Config.log.debug("\(try server["db_name"]["collection_name"].findOne())")
```

**Comment by Joannis** [comment]
I can't imagine this failing on our end, honestly. I'd need more details (verbose logging?) to say more.

**Comment by sujaykakkad** [comment]
I guess it was the problem with mongokitten3. 
I upgraded the version to mongokitten4.
It is working on mac but failing on linux.
I am getting `invalidBuildInfoDocument` error

**Comment by sujaykakkad** [comment]
The detailed error is here
```sh
fatal error: Error raised at top level: MongoKitten.MongoError.invalidBuildInfoDocument: file /home/buildnode/disk2/workspace/oss-swift-3.1-package-linux-ubuntu-16_04/swift/stdlib/public/core/ErrorType.swift, line 188
Current stack trace:
0    libswiftCore.so                    0x00007fb68d5271c0 swift_reportError + 120
1    libswiftCore.so                    0x00007fb68d541ad0 _swift_stdlib_reportFatalErrorInFile + 100
2    libswiftCore.so                    0x00007fb68d33c40c <unavailable> + 1188876
3    libswiftCore.so                    0x00007fb68d4d437d <unavailable> + 2859901
4    libswiftCore.so                    0x00007fb68d33bbe6 <unavailable> + 1186790
5    libswiftCore.so                    0x00007fb68d4da290 <unavailable> + 2884240
6    libswiftCore.so                    0x00007fb68d33c01f <unavailable> + 1187871
7    libswiftCore.so                    0x00007fb68d49afa9 <unavailable> + 2625449
8    libswiftCore.so                    0x00007fb68d33bbe6 <unavailable> + 1186790
9    libswiftCore.so                    0x00007fb68d458260 specialized _assertionFailure(StaticString, String, file : StaticString, line : UInt, flags : UInt32) -> Never + 144
10   libswiftCore.so                    0x00007fb68d364d15 <unavailable> + 1355029
11   MongoKittenTest                    0x0000000000418a41 <unavailable> + 100929
12   libc.so.6                          0x00007fb68adec740 __libc_start_main + 240
13   MongoKittenTest                    0x00000000004184a9 <unavailable> + 99497
Illegal instruction (core dumped)
```

**Comment by Joannis** [comment]
I've seen this before with a previous user. Whenever you're ready, you can poke me on slack. I'll need to know why the build info document is seen as invalid.

**Comment by Joannis** [comment]
Fixed in 4.0.10
